### PR TITLE
Update controller example code to Kubernetes 1.8

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: contributors/devel/controllers.md [hunk 1]
```diff
@@ -103,6 +103,30 @@ func (c *Controller) Run(threadiness int, stopCh chan struct{}) {
 		go wait.Until(c.runWorker, time.Second, stopCh)
 	}
 
+	// register event handlers to fill the queue with pod creations, updates and deletions
+	c.pods.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
```
Comment (liggitt) [inline contributors/devel/controllers.md]: this should happen during construction, not in Run(). adding the event handler here (after start) would be ignored

Comment (jhorwit2) [reply contributors/devel/controllers.md]: @liggitt it works after Run as been called. See [here](https://github.com/kubernetes/client-go/blob/db8228460e2de17f5d3a9a453f61dde0ba86545a/tools/cache/shared_informer.go#L318).

**Comment by castrojo** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/community/1122)

**Comment by spiffxp** [comment]
/retest
/lgtm
Just in time for 1.9

/hold
Is this still relevant given that we have https://github.com/kubernetes/sample-controller?  If you think so you're free to /hold cancel

**Comment by jhorwit2** [comment]
@spiffxp That's an example for CRD operators. The docs in this PR are for controllers. I think it's still relevant. I use this example all the time when writing new non-crd controllers.

**Comment by sttts** [comment]
> Is this still relevant given that we have https://github.com/kubernetes/sample-controller?

These are docs, sample-controller is an e2e example. Two purposes.

**Pull request merged**